packages/squiggle-lang/src/rescript/GenericDist/GenericDist_GenericOperation.res

```diff
@@ -0,0 +1,156 @@
+type operation = GenericDist_Types.Operation.genericFunctionCall
+type genericDist = GenericDist_Types.genericDist
+type error = GenericDist_Types.error
+
+// TODO: It could be great to use a cache for some calculations (basically, do memoization). Also, better analytics/tracking could go a long way.
+
+type params = {
+  sampleCount: int,
+  xyPointLength: int,
+}
+
+type outputType = [
+  | #Dist(genericDist)
+  | #Error(error)
+  | #Float(float)
+  | #String(string)
+]
+
+module Output = {
+  let toDist = (o: outputType) =>
+    switch o {
+    | #Dist(d) => Some(d)
+    | _ => None
+    }
+
+  let toFloat = (o: outputType) =>
+    switch o {
+    | #Float(d) => Some(d)
+    | _ => None
+    }
+
+  let toString = (o: outputType) =>
+    switch o {
+    | #String(d) => Some(d)
+    | _ => None
+    }
+
+  let toError = (o: outputType) =>
+    switch o {
+    | #Error(d) => Some(d)
+    | _ => None
+    }
+}
+
+let fromResult = (r: result<outputType, error>): outputType =>
+  switch r {
+  | Ok(o) => o
+  | Error(e) => #Error(e)
+  }
+
+let outputToDistResult = (b: outputType): result<genericDist, error> =>
+  switch b {
+  | #Dist(r) => Ok(r)
+  | #Error(r) => Error(r)
+  | _ => Error(ImpossiblePath)
+  }
+
+let rec run = (extra, fnName: operation): outputType => {
```

Comment:
``let rec run = (extra, fnName: operation)`` was made with the principle to have the key variable be last, so that it works well with pipes.

...However, now that I think about it, I've been used to the ``|>`` pipe, but Rescript seems to prefer the ``->`` pipe; in which case we might want to flip all of these.

Reply:
Apparently ``|>`` is deprecated. Sigh... that will be a substantial refactor. 
https://rescript-lang.org/docs/manual/latest/pipe#sidebar

Reply:
> Rescript seems to prefer the `->` pipe

Super good point! [the official story is that triangle pipe is _deprecated_](https://rescript-lang.org/docs/manual/latest/pipe#triangle-pipe-deprecated)

Reply:
I think I'll leave this as a TODO after this story goes in, it could take a bit of time, and the rest of our code uses "|>"

Reply:
It'll be on the board https://github.com/QURIresearch/squiggle/issues/123